dialer.go

```diff
@@ -13,6 +13,9 @@ type dialResult struct {
 	err error
 }
 
+// Dialer returns a GRPC net.Conn connected to the provided address
+//
+// The address must be for a unix socket
```

Comment:
It's technically a named pipe on windows. Maybe use `local transport` ?

Reply:
not in this method, look at the code

Reply:
Hum, I see. 

```go
address = strings.TrimPrefix(address, "unix://")
```

Should be moved into `dialer()` version of unix actually. `Dialer()` is used by all platforms.

Reply:
idk, maybe in another PR for the windows work

Reply:
I'll just make a quick PR to fix it -> #1558

Reply:
with #1558 basically ready to merge do you want to clarify the comment or handle via a follow-up PR @crosbymichael?